dGame/dComponents/VendorComponent.cpp

```diff
@@ -1,124 +1,96 @@
 #include "VendorComponent.h"
-
-#include <BitStream.h>
-
+#include "BitStream.h"
 #include "Game.h"
 #include "dServer.h"
-
 #include "CDComponentsRegistryTable.h"
 #include "CDVendorComponentTable.h"
 #include "CDLootMatrixTable.h"
 #include "CDLootTableTable.h"
+#include "CDItemComponentTable.h"
 
 VendorComponent::VendorComponent(Entity* parent) : Component(parent) {
+	m_HasStandardCostItems = false;
+	m_HasMultiCostItems = false;
 	SetupConstants();
 	RefreshInventory(true);
 }
 
-VendorComponent::~VendorComponent() = default;
-
 void VendorComponent::Serialize(RakNet::BitStream* outBitStream, bool bIsInitialUpdate, unsigned int& flags) {
-	outBitStream->Write1();
-	outBitStream->Write1(); // Has standard items (Required for vendors with missions.)
-	outBitStream->Write(HasCraftingStation()); // Has multi use items
+	outBitStream->Write(bIsInitialUpdate || m_DirtyVendor);
+	if (bIsInitialUpdate || m_DirtyVendor) {
+		outBitStream->Write(m_HasStandardCostItems);
+		outBitStream->Write(m_HasMultiCostItems);
+		if (!bIsInitialUpdate) m_DirtyVendor = false;
+	}
 }
 
 void VendorComponent::OnUse(Entity* originator) {
 	GameMessages::SendVendorOpenWindow(m_Parent, originator->GetSystemAddress());
 	GameMessages::SendVendorStatusUpdate(m_Parent, originator->GetSystemAddress());
 }
 
-float VendorComponent::GetBuyScalar() const {
-	return m_BuyScalar;
-}
-
-float VendorComponent::GetSellScalar() const {
-	return m_SellScalar;
-}
-
-void VendorComponent::SetBuyScalar(float value) {
-	m_BuyScalar = value;
-}
-
-void VendorComponent::SetSellScalar(float value) {
-	m_SellScalar = value;
-}
-
-std::map<LOT, int>& VendorComponent::GetInventory() {
-	return m_Inventory;
-}
+void VendorComponent::RefreshInventory(bool isCreation) {
+	SetHasStandardCostItems(false);
+	SetHasMultiCostItems(false);
 
-bool VendorComponent::HasCraftingStation() {
-	// As far as we know, only Umami has a crafting station
-	return m_Parent->GetLOT() == 13800;
-}
+	// Custom code for Max vanity NPC and Mr.Ree cameras
+	MaxCustomVendorRefreshInventory(isCreation);
+	if(m_Parent->GetLOT() == 9749 && Game::server->GetZoneID() == 1201) return;
```

Comment:
we should move the max method to be in this if statement and remove the if check in the other method since it should be private.